src/helpers/get-issues-data.ts

```diff
@@ -0,0 +1,138 @@
+import { FETCH_RANGED_COMMENTS } from "@/graphql/queries"
+import {
+    IssueCommentDataType,
+    IssueCommentNodes,
+    IssueComments
+} from "@/types/comments"
+
+export async function getIssueCommentsData({
+    username,
+    token,
+    query,
+    startDate,
+    endCursor
+}: {
+    username: string
+    token?: string
+    query: "FETCH_RANGED_COMMENTS"
+    startDate?: string
+    endCursor: string | null
+}) {
+    const tokenFromEnv = process.env.GITHUB_TOKEN
+
+    let graphqlQuery = ""
+    switch (query) {
+        case "FETCH_RANGED_COMMENTS":
+            graphqlQuery = FETCH_RANGED_COMMENTS
+            break
+        default:
+            throw new Error("Invalid query")
+    }
+
+    let hasNextPage = true
+
+    endCursor = startDate?.slice(0, 4) === "2022" ? endCursor : null
+
+    let startCursorObj: any = {}
+    let endCursorObj: any = {}
+
+    let allIssueComments: Array<IssueCommentNodes> = []
+
+    while (hasNextPage) {
+        if (
+            startDate?.slice(0, 4) !== "2023" &&
+            startDate?.slice(0, 4) !== "2022"
+        ) {
+            return {
+                data: [],
+                error: null
+            }
+        }
+
+        try {
+            const res = await fetch("https://api.github.com/graphql", {
+                method: "POST",
+                headers: {
+                    Authorization: `Bearer ${token ?? tokenFromEnv}`,
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify({
+                    query: graphqlQuery,
+                    variables: {
+                        username,
+                        endCursor
+                    }
+                })
+            })
+
+            if (!res.ok) {
+                return { error: res.statusText, message: "Failed to fetch API" }
+            }
+
+            const data = await res.json()
+            const jsonData: IssueCommentDataType = data.data.user
+
+            const nodes_data = jsonData?.issueComments?.nodes
+            const start_cursor = jsonData?.issueComments?.pageInfo?.startCursor
+            const isNextPage = jsonData?.issueComments?.pageInfo?.hasNextPage
+            const end_cursor = jsonData?.issueComments?.pageInfo?.endCursor
+
+            const prevYear = Number(startDate?.slice(0, 4)) - 1
```

Comment:
why are slicing just the 1st three values? put a comment here to explain why so its descriptive for anyone looking at the code